modules/core/src/test/java/org/apache/ignite/internal/processors/cache/GridEvictionPolicyMBeansTest.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License
+ */
+
+package org.apache.ignite.internal.processors.cache;
+
+import javax.management.ObjectName;
+import org.apache.ignite.cache.eviction.fifo.FifoEvictionPolicyFactory;
+import org.apache.ignite.cache.eviction.lru.LruEvictionPolicy;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.configuration.NearCacheConfiguration;
+import org.apache.ignite.internal.util.IgniteUtils;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+/**
+ * Tests for the eviction policy JMX beans registered by the kernal.
+ */
+public class GridEvictionPolicyMBeansTest extends GridCommonAbstractTest {
+
+    /** Create test and auto-start the grid */
+    public GridEvictionPolicyMBeansTest() {
+        super(true);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * This implementation registers adds custom executors to the configuration.
+     */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        FifoEvictionPolicyFactory<String, String> plc = new FifoEvictionPolicyFactory<>();
+        plc.setMaxSize(100);
+        plc.setBatchSize(10);
+        plc.setMaxMemorySize(20);
+
+        CacheConfiguration cache1 = defaultCacheConfiguration();
+        cache1.setName("cache1");
+        cache1.setOnheapCacheEnabled(true);
+        cache1.setEvictionPolicyFactory(plc);
+
+        NearCacheConfiguration ncf = new NearCacheConfiguration<>();
+        LruEvictionPolicy lep = new LruEvictionPolicy();
+        lep.setBatchSize(10);
+        lep.setMaxMemorySize(500);
+        lep.setMaxSize(40);
+        ncf.setNearEvictionPolicy(lep);
```

Comment:
Let's use `setNearEvictionPolicyFactory` here.

I think we need 4 cases here: `setEvictionPolicyFactory`, `setNearEvictionPolicyFactory`, `setEvictionPolicy` and `setNearEvictionPolicy`.

You can use factories in `cache1` and plain policies in `cache2`.